### ISSUE TITLE
FAIL: //tensorflow/contrib/seq2seq:beam_search_decoder_test 

### ISSUE THREAD

**Issue by lukeiwanski** [lukeiwanski/tensorflow#33]
## System Info

```
  Name:						 Hawaii
  Vendor:					 Advanced Micro Devices, Inc.
  Device OpenCL C version:			 OpenCL C 2.0 
  Driver version:				 1912.5 (VM)
  Profile:					 FULL_PROFILE
  Version:					 OpenCL 2.0 AMD-APP (1912.5)
  Extensions:					 cl_khr_fp64 cl_amd_fp64 cl_khr_global_int32_base_atomics cl_khr_global_int32_extended_atomics cl_khr_local_int32_base_atomics cl_khr_local_int32_extended_atomics cl_khr_int64_base_atomics cl_khr_int64_extended_atomics cl_khr_3d_image_writes cl_khr_byte_addressable_store cl_khr_gl_sharing cl_khr_gl_depth_images cl_ext_atomic_counters_32 cl_amd_device_attribute_query cl_amd_vec3 cl_amd_printf cl_amd_media_ops cl_amd_media_ops2 cl_amd_popcnt cl_khr_image2d_from_buffer cl_khr_spir cl_khr_subgroups cl_khr_gl_event cl_khr_depth_images cl_khr_mipmap_image cl_khr_mipmap_image_writes

```
ComputeCpp 0.2.0

## To reproduce
```
bazel test --config=sycl --test_tag_filters=local,-benchmark-test --local_test_jobs=4 -k --test_lang_filters=cc,py --action_env=LD_PRELOAD=/usr/lib/libOpenCL.so.1 --test_timeout 300,750,1200,3600 //tensorflow/contrib/seq2seq:beam_search_decoder_test
```

## Error
```
ERROR: test_step (__main__.TestBeamStep)
...
InvalidArgumentError: indices[0] = -1677676288 is not in [0, 30)
         [[Node: Gather = Gather[Tindices=DT_INT32, Tparams=DT_FLOAT, validate_indices=true, _device="/job:localhost/replica:0/task:0/device:SYCL:0"](Reshape_6, Reshape_5/_69)]]
         [[Node: Reshape_8/_51 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/device:SYCL:0", send_device_incarnation=1, tensor_name="edge_223_Reshape_8", tensor_type=DT_INT32, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

Caused by op u'Gather', defined at:
...
```

**Comment by lukeiwanski** [comment]
Should be fixed with 6a38b35c94b2ab88e2c861931fff6c09c3ae25ff

**Comment by lukeiwanski** [comment]
With the mentioned patch the error changed to:
```
Arrays are not equal

(mismatch 100.0%)
 x: array([[  2.802597e-45,   0.000000e+00,   0.000000e+00],
       [  0.000000e+00,   2.802597e-44,   0.000000e+00]], dtype=float32)
 y: array([[-1.318522, -1.880748, -2.080748],
       [-1.098804, -1.336722, -2.330112]], dtype=float32)
```

**Comment by jwlawson** [comment]
Fixed by #53

**Comment by lukeiwanski** [comment]
Awesome! Closing!
